packages/path_provider/path_provider_linux/test/get_application_id_test.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:path_provider_linux/src/get_application_id_real.dart';
+
+class FakeGioUtils implements GioUtils {
+  int? application = 0;
+  Pointer<Utf8>? applicationId = nullptr;
+
+  @override
+  bool libraryIsPresent = false;
+
+  @override
+  int gApplicationGetDefault() => application!;
+
+  @override
+  Pointer<Utf8> gApplicationGetApplicationId(int app) => applicationId!;
+}
+
+void main() {
+  final FakeGioUtils fakeGio = FakeGioUtils();
```

Comment:
Should this be in setup as well? We probably should avoid having this object stated in tests.

Reply:
Indeed! Done.